### ISSUE TITLE
PNG images are uploaded with names ending in .jpg

### ISSUE THREAD

**Issue by babolivier** [vector-im/element-x-android#1681]
### Steps to reproduce

1. Upload a PNG image
2. Look at the file name (either in the event source (`body`), or in Element Web/Desktop by hovering on the image)

#### What did you expect?

File name ends with `.png`

#### What happened instead?

File name ends with `.jpg`

### Your phone model

Fairphone 3

### Operating system version

Androind 13

### Application version and app store

Element X version 0.2.3 from the Play Store

### Homeserver

abolivier.bzh

### Will you send logs?

Yes

### Are you willing to provide a PR?

No

**Comment by babolivier** [comment]
Also heads up the issue template's instructions on uploading logs and finding the app version are incorrect.

**Comment by jmartinesp** [comment]
Non-animated image files sent as media (with the photo gallery picker) will be automatically resized and compressed to a JPG, so this is expected. If you want to send the image file as is, you should select the 'Attachment' option instead. 

We might want to add some option to disable/customize this behavior in the future.

**Comment by babolivier** [comment]
This makes sense, however when I download the file it downloads as a PNG - and the MIME type in the event is also `image/png`.

**Comment by jmartinesp** [comment]
Ok, that definitely looks like a bug 😅 .

**Comment by ibrahim-iqbal** [comment]
I traced the upload path against `develop` and want to flag what I see before opening a PR, because it looks like several code paths have moved since this was reported and the remaining exposure isn't quite what the original repro describes.

The gallery-picker path today:

- `AndroidMediaPreProcessor.process()` reads the URI's declared mime type. `resolveImageMimeType` at `libraries/mediaupload/impl/src/main/kotlin/io/element/android/libraries/mediaupload/impl/AndroidMediaPreProcessor.kt:165` short-circuits when the declared type is already `image/png` or `image/jpeg`, otherwise it sniffs the image header via `BitmapFactory.Options.outMimeType`.
- `ImageCompressor.compressToTmpFile` picks the compressor format and file extension via `mimeTypeToCompressFileExtension` in `MimeTypeUtil.kt`, which returns `"png"` for PNG and `"jpeg"` otherwise. The tmp file is created with that extension.
- `postProcess` at `AndroidMediaPreProcessor.kt:146-159` renames the tmp file back to `context.getFileName(uri)` — the picker's `DISPLAY_NAME` for content URIs, or `Uri.toFile().name` for file URIs — so the extension on the outgoing file matches whatever the picker reported. For a straight PNG pick from the gallery this should preserve `.png`.

Where the `.jpg` name still comes out is more specific than "any PNG upload":

- The camera-capture flow hardcodes `getTemporaryFile("photo.jpg")` at `libraries/mediapickers/impl/src/main/kotlin/io/element/android/libraries/mediapickers/impl/DefaultPickerProvider.kt:148`, so any bitmap the camera writes to that file goes out as `.jpg` in the message body, regardless of what the camera actually encoded. That is the flow @rugk's Fairphone 3 hits when opening an OS-provided camera intent from the composer.
- If a share intent hands us a content URI whose `DISPLAY_NAME` cursor is null or lacks an extension, `postProcess` returns `this` and the tmp file keeps its compressor-assigned name (`<uuid>.jpeg`) — so a PNG shared without a real display name lands as `.jpeg`, not `.jpg`.

Wanted to check the shape you'd prefer before writing anything:

1. Rename the camera capture target based on the actual encoded mime type (read from the resulting bytes) rather than always `photo.jpg`. This covers the reproducible case on modern devices where the camera does write PNG. Small change; targeted at `DefaultPickerProvider.registerCameraPhotoPicker` and its callers.
2. Fall back to the compressor's extension in `postProcess` when the source URI has no usable name, so the outgoing file at least matches the encoded bytes. Also small, and independent of (1).
3. Both.

Could this be assigned to me while I put (1) or (1+2) together? Happy to do it as two separate PRs if you'd rather keep the camera and share paths distinct.

For disclosure: this contribution is being drafted with AI assistance.

**Comment by rugk** [comment]
> this contribution is being drafted with AI assistance.

And it apparently invented my involvement(?). 🤔 

Anyway, interesting issue though.
